client/maps/browse-server-maps.jsx

```diff
@@ -285,12 +282,10 @@ export default class Maps extends React.Component {
     this.props.dispatch(clearMapsList())
     // Since we're using the same instance of the InfiniteList component to render maps, gotta
     // reset it each time we change tabs, or apply different filters. Also, make sure to do it
-    // after resetting the `currentPage` to 0, since the infinite list will start to load maps
-    // with whatever is currently saved in state.
-    this.setState(
-      () => ({ currentPage: 0 }),
-      () => this.infiniteList.reset(),
-    )
+    // before resetting the `currentPage` to 0, since the refresh token is a regular class field
+    // which doesn't cause a re-render.
+    this._refreshToken++
+    this.setState(() => ({ currentPage: 0 }))
```

Comment:
nit: this can just be the passed state form of `setState` instead of the callback form now